src/hooks.ts

```diff
@@ -57,7 +64,39 @@ export function useEllipsePosition(props: UseEllipsePositionConfig) {
     ry: convertValueAbs(ry, plotHeight, yScale),
   };
 }
-
+export function useDirectedEllipsePosition(
+  props: UseDirectedEllipsePositionConfig,
+) {
+  const { axisContext, plotWidth, plotHeight } = usePlotContext();
+  const [xScale, yScale] = validateAxis(axisContext, 'x', 'y');
+  const { x1: oldX1, y1: oldY1, x2: oldX2, y2: oldY2, width } = props;
+  const { x1, y1, x2, y2 } = {
+    x1: convertValue(oldX1, plotWidth, xScale),
+    x2: convertValue(oldX2, plotWidth, xScale),
+    y1: convertValue(oldY1, plotWidth, yScale),
+    y2: convertValue(oldY2, plotWidth, yScale),
+  };
+  const radsToDegs = (rad: number) => (rad * 180) / Math.PI;
+  const { cx, cy } = {
+    cx: (x1 + x2) / 2,
+    cy: (y1 + y2) / 2,
+  };
+  return {
+    cx,
+    cy,
+    rx: distanceXY(x1, y1, x2, y2) / 2,
+    ry: convertValueAbs(width, plotHeight, yScale) / 2,
+    rotation:
+      (y1 > y2 ? -1 : 1) *
+      (x1 > x2 ? -1 : 1) *
+      radsToDegs(
+        Math.asin(distanceXY(x1, y1, x1, cy) / distanceXY(x1, y1, cx, cy)),
+      ),
+  };
+}
+function distanceXY(x1: number, y1: number, x2: number, y2: number) {
```

Comment:
You can use the `ml-euclidean-distance` which is already a dependency in this project

Reply:
fixed